Helper/ApiClient.php

```diff
@@ -64,51 +63,64 @@ public function __construct(
     }
 
     /**
+     * @param bool $refresh
      * @return int
      */
-    protected function getCurlStatus()
+    protected function getCurlStatus($refresh = false)
     {
-        if ($this->_status === null) {
+        if ($this->_status === null || $refresh) {
             $this->_status = $this->_curl->getStatus();
         }
 
         return $this->_status;
     }
 
     /**
+     * @param bool $refresh
      * @return array
      */
-    protected function getCurlHeaders()
+    protected function getCurlHeaders($refresh = false)
     {
-        if ($this->_headers === null) {
+        if ($this->_headers === null || $refresh) {
             $this->_headers = $this->_curl->getHeaders();
         }
 
         return $this->_headers;
     }
 
     /**
+     * @param bool $refresh
      * @return array
      */
-    protected function getCurlBody()
+    protected function getCurlBody($refresh = false)
     {
-        if ($this->_body === null) {
+        if ($this->_body === null || $refresh) {
             $this->_body = json_decode($this->_curl->getBody());
         }
 
         return $this->_body;
     }
 
+    /**
+     * @return $this
+     */
+    protected function clearResponseData()
+    {
+        $this->_body = $this->_status = $this->_headers = null;
+        return $this;
+    }
+
     /**
      * @return array
      */
     protected function prepareCurlResponseData()
     {
-        return [
+        $responseData = [
             'status' => $this->getCurlStatus(),
             'headers' => $this->getCurlHeaders(),
             'body' => $this->getCurlBody(),
         ];
+        return $responseData;
     }
 
     protected function isOkResponse()
```

Comment:
I don't see any use of this function in the project... is it redundant?

Reply:
It's a generic function for CURL helpers. Not in use at the moment, but might be useful in the future in order to replace lines like `if ($result['status'] == 200 ...` with prettier code.
You may remove it, but I would keep it anyway.

Reply:
I think it's better to remove it, and if we will need it in the future, we can re-add it.